drivers/bluetooth/hci/userchan.c

```diff
@@ -93,6 +93,53 @@ static struct net_buf *get_rx(const uint8_t *buf)
 	return NULL;
 }
 
+/**
+ * @brief Decode the length of an HCI H4 packet
+ * @details Decodes packet length according to Bluetooth spec v5 Vol 4 Part E
+ * @param packet_type	Type of packet according to Bluetooth spec v5.2 Vol 4, Part A Table 2.1
+ * @param packet_buff	Pointer to packet buffer
+ */
+static uint16_t decode_hci_packet_length(uint8_t packet_type, uint8_t *packet_buff)
+{
+	uint16_t payload_size = 0;
+	uint8_t header_size = 0;
+	const uint8_t packet_type_len = 1;
+
+	switch (packet_type) {
+	/* Bluetooth spec 5.2 Vol 4, Part E - 5.4.1 HCI Command Packet */
+	case H4_CMD:
+		payload_size = packet_buff[3];
+		header_size = 3;
+		break;
```

Comment:
There is some information missing from when you are doing `[3]` or `[2]`. I assume that is has something to do with the header sizes? 

Suggest to use defined values, rather than magic numbers, to retrieve the payload sizes